cpp/include/cucim/plugin/image_format.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUCIM_PLUGIN_IMAGE_FORMAT_H
+#define CUCIM_PLUGIN_IMAGE_FORMAT_H
+
+#include "cucim/filesystem/file_path.h"
+#include "cucim/io/format/image_format.h"
+
+
+namespace cucim::plugin
+{
+
+class ImageFormat
+{
+public:
+    ImageFormat() = default;
+    ~ImageFormat() = default;
+
+    bool add_interfaces(cucim::io::format::IImageFormat* image_formats);
```

Comment:
Is it possible to use a reference here? Could the variable be `const`?

Reply:
Thanks for pointing it out!
Updated to use `const` variable.